barretenberg/cpp/src/barretenberg/vm/avm/generated/full_row.hpp

```diff
@@ -6,781 +6,40 @@
-    FF perm_pos_mem_write_d_inv{};
-    FF perm_pos2_fixed_pos2_perm_inv{};
-    FF perm_slice_mem_inv{};
-    FF perm_merkle_poseidon2_inv{};
-    FF perm_main_alu_inv{};
-    FF perm_main_bin_inv{};
-    FF perm_main_conv_inv{};
-    FF perm_main_sha256_inv{};
-    FF perm_main_pos2_perm_inv{};
-    FF perm_main_mem_a_inv{};
-    FF perm_main_mem_b_inv{};
-    FF perm_main_mem_c_inv{};
-    FF perm_main_mem_d_inv{};
-    FF perm_main_mem_ind_addr_a_inv{};
-    FF perm_main_mem_ind_addr_b_inv{};
-    FF perm_main_mem_ind_addr_c_inv{};
-    FF perm_main_mem_ind_addr_d_inv{};
-    FF lookup_rng_chk_pow_2_inv{};
-    FF lookup_rng_chk_diff_inv{};
-    FF lookup_rng_chk_0_inv{};
-    FF lookup_rng_chk_1_inv{};
-    FF lookup_rng_chk_2_inv{};
-    FF lookup_rng_chk_3_inv{};
-    FF lookup_rng_chk_4_inv{};
-    FF lookup_rng_chk_5_inv{};
-    FF lookup_rng_chk_6_inv{};
-    FF lookup_rng_chk_7_inv{};
-    FF lookup_mem_rng_chk_0_inv{};
-    FF lookup_mem_rng_chk_1_inv{};
-    FF lookup_mem_rng_chk_2_inv{};
-    FF lookup_pow_2_0_inv{};
-    FF lookup_pow_2_1_inv{};
-    FF lookup_byte_lengths_inv{};
-    FF lookup_byte_operations_inv{};
-    FF lookup_opcode_gas_inv{};
-    FF lookup_l2_gas_rng_chk_0_inv{};
-    FF lookup_l2_gas_rng_chk_1_inv{};
-    FF lookup_da_gas_rng_chk_0_inv{};
-    FF lookup_da_gas_rng_chk_1_inv{};
-    FF lookup_cd_value_inv{};
-    FF lookup_ret_value_inv{};
-    FF incl_main_tag_err_inv{};
-    FF incl_mem_tag_err_inv{};
-    FF lookup_rng_chk_pow_2_counts{};
-    FF lookup_rng_chk_diff_counts{};
-    FF lookup_rng_chk_0_counts{};
-    FF lookup_rng_chk_1_counts{};
-    FF lookup_rng_chk_2_counts{};
-    FF lookup_rng_chk_3_counts{};
-    FF lookup_rng_chk_4_counts{};
-    FF lookup_rng_chk_5_counts{};
-    FF lookup_rng_chk_6_counts{};
-    FF lookup_rng_chk_7_counts{};
-    FF lookup_mem_rng_chk_0_counts{};
-    FF lookup_mem_rng_chk_1_counts{};
-    FF lookup_mem_rng_chk_2_counts{};
-    FF lookup_pow_2_0_counts{};
-    FF lookup_pow_2_1_counts{};
-    FF lookup_byte_lengths_counts{};
-    FF lookup_byte_operations_counts{};
-    FF lookup_opcode_gas_counts{};
-    FF lookup_l2_gas_rng_chk_0_counts{};
-    FF lookup_l2_gas_rng_chk_1_counts{};
-    FF lookup_da_gas_rng_chk_0_counts{};
-    FF lookup_da_gas_rng_chk_1_counts{};
-    FF lookup_cd_value_counts{};
-    FF lookup_ret_value_counts{};
-    FF incl_main_tag_err_counts{};
-    FF incl_mem_tag_err_counts{};
+template <typename FF_> struct AvmFullRow {
+    using FF = FF_;
 
-    RefVector<const FF> as_vector() const;
+    FF ALL_ENTITIES;
 
+    RefVector<const FF> as_vector() const;
     static std::vector<std::string> names();
     static constexpr size_t SIZE = 764;
+
+    // Risky but oh so efficient.
+    FF& get_column(ColumnAndShifts col)
+    {
+        static_assert(sizeof(*this) == sizeof(FF) * static_cast<size_t>(ColumnAndShifts::NUM_COLUMNS));
+        return reinterpret_cast<FF*>(this)[static_cast<size_t>(col)];
+    }
```

Comment:
"Risky" because I'm reinterpreting the struct as a bunch of consecutive FFs. This could or could not be true, depending on what the compiler decides to do, padding, etc. Given FFs are 32 bits, it's likely to be a packed struct (so, yes, consecutive FFs). I have the static assert to make sure of that.

Efficient because it lets us access the row both by member `row.some_column` and by index `row.get_colum(index)`. This allows many optimizations and we don't need to codegen as much.

However, fear not, in the new design the full row is only used for testing!